sway-core/src/transform/to_parsed_lang/convert_parse_tree.rs

```diff
@@ -2044,6 +2045,199 @@ fn expr_func_app_to_expression_kind(
+                            current_file, current_line_col.start.line, current_line_col.start.col
+                        ),
+                        &span,
+                    ),
+                    // arg.fmt(f);
+                    AstNode {
+                        content: AstNodeContent::Expression(Expression {
+                            kind: ExpressionKind::MethodApplication(Box::new(
+                                MethodApplicationExpression {
+                                    method_name_binding: TypeBinding {
+                                        inner: MethodName::FromModule {
+                                            method_name: BaseIdent::new_no_span("fmt".into()),
+                                        },
+                                        type_arguments: TypeArgs::Regular(vec![]),
+                                        span: Span::dummy(),
+                                    },
+                                    contract_call_params: vec![],
+                                    arguments: vec![
+                                        Expression {
+                                            kind: ExpressionKind::Variable(arg_ident.clone()),
+                                            span: Span::dummy(),
+                                        },
+                                        Expression {
+                                            kind: ExpressionKind::Variable(f_ident.clone()),
+                                            span: Span::dummy(),
+                                        },
+                                    ],
+                                },
+                            )),
+                            span: Span::dummy(),
+                        }),
+                        span: Span::dummy(),
+                    },
+                    // f.print_str(<newline>);
+                    ast_node_to_print_str(f_ident.clone(), "\n", &span),
+                    // arg
+                    AstNode {
+                        content: AstNodeContent::Expression(Expression {
+                            kind: ExpressionKind::ImplicitReturn(Box::new(Expression {
+                                kind: ExpressionKind::AmbiguousVariableExpression(
+                                    arg_ident.clone(),
+                                ),
+                                span: Span::dummy(),
+                            })),
+                            span: Span::dummy(),
+                        }),
+                        span: Span::dummy(),
+                    },
+                ],
+                whole_block_span: Span::dummy(),
+            };
+
+            return Ok(ExpressionKind::CodeBlock(block));
+        }
+        // ... and in release becomes "arg"
+        Some(Intrinsic::Dbg)
+            if !context.is_dbg_generation_full() && last.is_none() && !is_relative_to_root =>
+        {
+            return Ok(arguments[0].kind.clone());
```

Comment:
Here we should also check for `arguments.len() != 1` and emit error. Otherwise, if we build empty call `__dbg()` in release the compiler will panic.

Since the two match arms differ only in `[!]context.is_dbg_generation_full()` and both need the check for arguments length, it will be simpler to have only one match arm, the check only on top of it and then the `if context.is_dbg_generation_full()` condition in the body.

Reply:
Good catch. I am also creating one test for empty `__dbg()`.